### ISSUE TITLE
Possible to utilize slideshow controls from nextcloud/viewer?

### ISSUE THREAD

**Issue by tophneal** [nextcloud/files_pdfviewer#444]
### Steps to reproduce
1. Enable Files viewer and Pdf viewer apps
2. Navigate to folder of PDFs
3. Open PDF in viewer lightbox

### Expected behaviour
(More desired than expected behavior) Utilize same left/right controls that files viewer uses for photos/videos.

### Actual behaviour
No controls, can only open a single PDF at a time.

### Server configuration

**Operating system**:
Debian

**Web server:**
Apache2

**Database:**
MySql 10.3.30

**PHP version:**
7.4.19

**Nextcloud version:** (see Nextcloud admin page)
21.0.2

**Where did you install Nextcloud from:**
cloudamo script

(Excluding the rest of the info, as this is not an issue, but a feature request.)

Is this feature request a possibility? I understand from reading the initial issue submitted for this app to use nextcloud/viewer there may be limitations.

**Comment by skjnldsv** [comment]
@jancborchardt does that make sense? I'm not sure tbh. Switching pdf like photos?

**Comment by tophneal** [comment]
> @jancborchardt does that make sense? I'm not sure tbh. Switching pdf like photos?

Precisely what I meant, yes! Sorry for any confusion in my explanation.

**Comment by piaaminda** [comment]
Any news on this? We are looking for exactly this feature

**Comment by tophneal** [comment]
Found a better way to get what I was looking for. Came across this [issue](https://github.com/nextcloud/viewer/issues/229#issuecomment-859667708) when troubleshooting errors from previews and used the advice to add "application/pdf", to the image mimetypes in /apps/viewer/viewer-main.js

Now the native file viewer loads pdfs and provide carousel controls to easily move to the next pdf!

**Comment by gpz1100** [comment]
@tophneal Can you clarify where and what exactly you're adding to the /apps/viewer/viewer-main.js file?

Not familiar at all with javascript, but looked at that file in notepad++, parsed through jstool (so it does the indents and all). Completely lost where to find the image mimetypes section.  THanks!

**Comment by tophneal** [comment]
@gpz1100 yes, I opened the file in vscode then searched for ‘mime’ until I reached a list with items like ‘image/jpg’ and add the ‘application/pdf’, to the end of the list/array.

**Comment by gpz1100** [comment]
@tophneal 

I don't have viewer-main.js in .../apps/viewer, I have it in /var/www/nextcloud/apps/viewer/js. It's a pretty large file, 1,270,150 bytes.

It does not contain a line "image/jpg".  Nextcloud 22.2.0 here.

**Comment by tophneal** [comment]
@gpz1100 

sorry for confusion, my NC is setup on a VPS, so locations may be a bit different. I’m also on v23, but idk if that affects this much or at all. 

the folder you’re seeing would be the correct one. I placed mine in the file, around line 49109 for me. It should go into here:

Id: “images”,
Group: “media”,
Mimes: [ … ],

It’ll go in the list within the brackets.

**Comment by gpz1100** [comment]
Ok. Found the right spot by searching for image/png. I added "application/pdf" in the correct format as the rest of the entries. I now get an error when trying to open any kind of pdf file.

https://i.imgur.com/Wn9MLas.png

**Comment by tophneal** [comment]
Huh. I wonder if it works on mine because I also added PDFs in my config for preview generation. Only other thing I did was refresh my page after making sure the js file was formatted correctly and refreshing the page.

**Comment by gpz1100** [comment]
Any updates from the nextcloud community on this?  Version 24 of nextcloud still requires one to open/close pdf's individually.

**Comment by skjnldsv** [comment]
Hey!
Sorry, I don't think this is something that we want.
Let me ask @jancborchardt and @nimishavijay again :)

**Comment by jancborchardt** [comment]
This is possible e.g. in both preview on macOS, as well as in Google Drive. It is a nice quick way of jumping between documents if you want to check, and it would be good for us to have.

(In general it would also be good to look at the PDF viewer of Google Drive, it looks very sleek and would fit our style much more too.)

**Comment by gpz1100** [comment]
> Hey! Sorry, I don't think this is something that we want. Let me ask @jancborchardt and @nimishavijay again :)

Why wouldn't you want this? Where's the efficiency of having to do multiple clicks to open the next/previous file?

**Comment by skjnldsv** [comment]
> > Hey! Sorry, I don't think this is something that we want. Let me ask @jancborchardt and @nimishavijay again :)
> 
> Why wouldn't you want this? Where's the efficiency of having to do multiple clicks to open the next/previous file?

Personal opinion :shrug: 
In any case, it's been approved!
